### PR TITLE
make embed & initializer coverage 100%

### PR DESCRIPTION
**Pull request by chris-chris**
Implemented some unit tests for embed.py and initializer.py.

스크린샷 2020-02-23 오전 4 02 19

### CODE REVIEW

File: haiku/_src/embed_test.py [hunk 1]
```diff
@@ -73,6 +73,40 @@ def test_inconsistent_creation_args(self):
...
+  def test_embed_invalid_lookup(self, lookup_style, inp_ids):
+    emb = embed.Embed(embedding_matrix=_EMBEDDING_MATRIX,
+                      lookup_style=lookup_style)
+    with self.assertRaisesRegex(ValueError, f"{lookup_style} is not a valid enum in EmbedLookupStyle."):
+      emb(inp_ids)
+
+  @parameterized.parameters(
+    itertools.product(["ONE_HOT"]))
+  @test_utils.transform_and_run
+  def test_embed_property_check(self, lookup_style):
```
Comment (tomhennigan) [inline haiku/_src/embed_test.py]: There is only one value for `lookup_style` here, I would suggest removing the @parameterized and method parameter and passing this directly.

File: haiku/_src/initializers_test.py [hunk 2]
```diff
@@ -61,5 +62,45 @@ def test_initializers(self, dtype):
       self.assertEqual(generated.shape, shape)
       self.assertEqual(generated.dtype, dtype)
 
+  @test_utils.transform_and_run
+  def test_invalid_variance_scale(self):
+
+    with self.assertRaises(ValueError):
```
Comment (tomhennigan) [inline haiku/_src/initializers_test.py]: Please test the error message is as expected here. For example:

```python
with self.assertRaisesRegex(ValueError, "scale.*must be a positive float"):
  initializers.VarianceScaling(scale=-1.0)
```

Same for the two below.

File: haiku/_src/embed_test.py [hunk 3]
```diff
@@ -73,6 +73,39 @@ def test_inconsistent_creation_args(self):
     with self.assertRaisesRegex(ValueError, "supplied but the `embed_dim`"):
       embed.Embed(embedding_matrix=_EMBEDDING_MATRIX, embed_dim=5)
 
+  @parameterized.parameters(
+    itertools.product(["ARRAY_INDEX", "ONE_HOT"]))
+  @test_utils.transform_and_run
```
Comment (trevorcai) [inline haiku/_src/embed_test.py]: No need for itertools.product here, or in `test_embed_invalid_lookup`.

File: haiku/_src/initializers_test.py [hunk 4]
```diff
@@ -61,5 +62,47 @@ def test_initializers(self, dtype):
...
+    fan_in_out3 = initializers._compute_fans([3, 4])
+    self.assertEqual(fan_in_out3, (3, 4))
+    fan_in_out4 = initializers._compute_fans([1, 2, 3, 4])
+    self.assertEqual(fan_in_out4, (6, 8))
+
+  @parameterized.parameters(np.float32, jnp.float32)
+  @test_utils.transform_and_run
+  def test_orthogonal_invalid_shape(self, dtype):
+    init = initializers.Orthogonal()
+    shape = (20, )
```
Comment (trevorcai) [inline haiku/_src/initializers_test.py]: nit: no trailing space after comma in one-tuples

Comment (chris-chris) [reply haiku/_src/initializers_test.py]: Done! Thanks!

File: haiku/_src/embed_test.py (hunk 3 above)
Comment (chris-chris) [reply haiku/_src/embed_test.py]: Ah- I agree with that. Fixed!

**Comment by chris-chris** [comment]
@trevorcai 
I downloaded the pylintrc from here.
https://www.tensorflow.org/community/contribute/code_style
https://raw.githubusercontent.com/tensorflow/tensorflow/master/tensorflow/tools/ci_build/pylintrc

I ran the pylint, and fixed it. Thank you for the comments!

**Pull request merged**